src/porepy/numerics/ad/grid_operators.py

```diff
@@ -557,6 +563,86 @@ def __str__(self) -> str:
         return s
 
 
+class Geometry(Operator):
+    """Wrapper class for Ad representations of grids.
+
+
+    Attributes:
+        cell_volumes (pp.ad.Matrix): Diagonal ad matrix of cell volumes.
+        face_areas (pp.ad.Matrix):  Diagonal ad matrix of face areas.
+        nd (int): Ambient/highest dimension of the mixed-dimensional grid.
+
+    FIXME: Implement parse??
+    """
+
+    def __init__(self, subdomains: List[pp.Grid], nd: int, name: Optional[str] = None):
+        """Construct concatenated grid operators for a given set of subdomains.
+
+        The operators will be ordered according to the ordering in grids. It is critical
+        that the same ordering is used by other operators.
+
+        Parameters:
+            subdomains (List of pp.Grid): List of grids. The order of the grids in the list
+                sets the ordering of the geometry operators. Can be either subdomain (pp.Grid)
+                or interface (pp.MortarGrid) grids.
+            nd: ambient dimension.
+            name (str, optional): Name of the operator. Default is None.
+
+        """
+        super().__init__(name=name)
+
+        self.subdomains = subdomains
+        self._num_grids: int = len(subdomains)
+        self.nd = nd
+
+        self.num_cells: int = sum([g.num_cells for g in subdomains])
+        self.num_faces: int = sum([g.num_faces for g in subdomains])
+
+        # Wrap the stacked matrices into Ad objects (could be extended to e.g. face normals)
+        for field in ["cell_volumes", "face_areas"]:
+            ad_matrix = Matrix(
+                sps.diags(np.hstack([getattr(g, field) for g in subdomains]))
+            )
+            setattr(self, field, ad_matrix)
+
+        def scalar_to_nd(size):
+            """Expand matrix of size [N x M] to [nd*N x M].
+
+            When left multiplied to a matrix A, each row of A is
+            repeated nd times.
+
+            Usage example: Scaling from traction to force is
+
+                    force = (scalar_to_nd_face * face_areas) * traction
+
+            FIXME: Refactor?
+            """
+            rows: np.ndarray = np.arange(size * self.nd)
+            cols: np.ndarray = np.kron(np.arange(size), np.ones(self.nd))
+            data: np.ndarray = np.ones(size * self.nd)
+            mat = sps.csr_matrix(
+                (data, (rows, cols)),
+                shape=(size * self.nd, size),
+            )
+            return pp.ad.Matrix(mat)
+
+        self.scalar_to_nd_cell = scalar_to_nd(self.num_cells)
+        self.scalar_to_nd_face = scalar_to_nd(self.num_faces)
+
+    def __repr__(self) -> str:
+        s = (
+            f"Geometry operator for {self._num_grids} grids.\n"
```

Comment:
I'm not sure 'Geometry operator' is the best word for this, but I have no better suggestions. Perhaps it is mainly a question of getting used to it.